### DISCUSSION TITLE
container-dtc-nwp version 3.5 released

### DISCUSSION THREAD

**Discussion by jwolff-ncar** [NCAR/container-dtc-nwp#68] [Announcements]
Along with the new code base, the [online tutorial documentation](https://dtcenter.org/nwp-containers-online-tutorial) was updated accordingly.

**Comment by lohancock** [comment]
Positive feedback here.  It is working beautifully so far.   Just to share some lessons learned - 

(1) So far it is working OK to use spot instances to save money - just start with spot request, in that dialogue tell it to search for the AMI, name that as dtc-utlity-base-env and say it is in Community AMIs.  Next to choose the instance type - it's not possible to ask for a c4.4xlarge with extra memory - you have to pick something without customization - so - it worked to choose "Specify instance attributes that match your compute requirements" and there say CPUs should be at least 16 and no more than 20, memory 60 GiB to 75 but no more.  This provides a pool that is expected to charge at 37 cents/hour, which is less than 80 cents an hour....  I know, I know, spot instances are risky.  but for a tutorial..
  
(2) Also working OK to use Instance Connect instead of Putty.  It is easier, especially since I am using a Chromebook.  In this case you need HTTP security rule that admits traffic from the Instance Connect server, not MyIP - and I couldn't find that IP - so I had to throw open HTTP to the world.  Not very secure but ... for a tutorial... it's easier...

Of interest, Instance Connect suggested i log in as root, but I used ec2-user since that seems to be what the tutorial uses.  This gave no trouble at any time.

(3) In many cases I had to type in the commands that are provided on the tutorial page - I could not cut and paste, I had to type them in.  No Problem - it worked - but fyi it appears the font being used on the tutorial page does not have a counterpart that is adequately similar, and so cut and paste will not work.  Though maybe I am the only one going by Chromebook :) so the only one who cared about this.

Just type it all in yourself, do not copy/paste.  And remember that ` is the quote mark on the top line with the tilde, not the one with the double quote mark. 

and then it works, smooth as silk.

(4)  After all the input data is downloaded, I ran ls -ald -- *  and of interest, the file information echoed back is not exactly same as what is listed on the tutorial web page.  

Permissions for the gsi folder are echoed back as drwxrwxr-x , not drwxr-xr-x. 
And for owner and group - the tutorial shows those fields as user admin - but i am getting ec2-user ec2-user

(5) When I pull the docker images and convert them to singularity, one by one, everything works (and for some reason I do not have to type it in this time - did you use a different font?) - with an important exception - the python page indicates that the directory listing after this step should include ncl_latest.sif but it does not.

And tomorrow I will try out the cases. 

This is really terrific and so far as I can tell it is working perfectly.  Thank you so much!
